xdsl/dialects/hw.py

```diff
@@ -24,6 +26,34 @@
 from xdsl.printer import Printer
 
 
+@dataclass
+class InnerSymTarget:
+    """The target of an inner symbol, the entity the symbol is a handle for."""
+
+    op: Operation | None = None
```

Comment:
It represents an invalid target, which is falsey when used as bool. It is returned from `lookup()` functions in `InnerSymbolTable` when no matching operation is found.

I’m not sure exactly why that’s preferred to an optional `InnerSymTarget` with a mandatory `Operation`, but that’s the way it was introduced upstream in [39c1805](https://www.github.com/llvm/circt/commit/39c180552b8ff7f7b598f7c7c1da94c923a8563b).

I’m guessing it could be so that the mapping from a looked-up name to a target object is unique, allowing the returned object to be modified to point to the right target at a later time. (While `InnerSymTarget` has no setter for `op`, the C++ code defines copy assignment operators.) I’ve got no concrete examples of this, but it may happen in other dialects such as FIRRTL.

Reply:
Can you please link the lookup function in the doc string for this class, and add a bit of this text in there? Seems like a good thing to have easily available for future readers of this code.